components/organisms/PhaseBanner.js

```diff
@@ -0,0 +1,297 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useTranslation } from "next-i18next";
+import { ActionButton } from "../atoms/ActionButton";
+import { useState } from "react";
+import Joi from "joi";
+import { ErrorLabel } from "../atoms/ErrorLabel";
+
+/**
+ * Displays the PhaseBanner on the page
+ */
+
+export const PhaseBanner = ({ phase, children, feedbackActive }) => {
+  const { t } = useTranslation("common");
+  const [ariaExpanded, setAriaExpanded] = useState(false);
+  const [caretHtml, setCaretHtml] = useState("&#9656;");
+  const [submitted, setSubmitted] = useState(false);
+
+  // Joi form validation schema.
+  const formSchema = Joi.object({
+    feedback: Joi.string()
+      .required()
+      .error((errors) => {
+        errors.forEach((error) => {
+          switch (error.code) {
+            case "any.required":
+              error.message = t("feedbackRequired");
+              break;
+            default:
+              error.message = t("feedbackRequired");
+              break;
+          }
+        });
+        return errors;
+      }),
+  });
+
+  const [feedback, setFeedback] = useState("");
+  const [feedbackError, setFeedbackError] = useState("");
+
+  let onSubmitHandler = async (e) => {
+    // prevent default behaviour of form
+    e.preventDefault();
+    // clear out error values
+    await setFeedbackError("");
+    // compile data into one object
+    const formData = { feedback };
+    // validate data using Joi schema
+    const { error } = formSchema.validate(formData, {
+      abortEarly: false,
+      allowUnknown: true,
+    });
+    console.log(formData);
+    console.log(error);
+    const valid = error === undefined;
+
+    if (valid) {
+      // create URLSearchParams object from FormData object
+      // this will be used to create url encoded string of names and values of the form fields
+      const urlEncoded = new URLSearchParams(formData);
+      // call feedback API route
+      fetch("/api/feedback", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+        },
+        body: urlEncoded.toString(),
+      }).catch((e) => {
+        // handle error if fetch fails
+        // fetch only fails if there is no internet connection not for the actual
+        // request so there is nothing really to do here other than to log it
+        console.log(e);
+      });
+
+      setSubmitted(true);
+    } else {
+      setFeedbackError(error.message);
+    }
+  };
+
+  //Function for changing feedback button state
+  function onfeedbackClick() {
+    const feedbackDropdown = document.getElementById("feedbackDropdown");
+
+    feedbackDropdown.classList.toggle("active");
+    ariaExpanded === true ? setAriaExpanded(false) : setAriaExpanded(true);
+
+    ariaExpanded === false ? setCaretHtml("&#9660;") : setCaretHtml("&#9656;");
+  }
```

Comment:
I'm still not sure I like how this is being done, but we can discuss it. You're using aria attributes and I'm not super familiar with them. I would have done this similar to how the ReportAProblem component is done though, where it's either on the page, or not there are all, instead of toggling visibility. Of course, if there are a11y issues with the way the ReportAProblem component is doing it, then it makes sense not to do it that way.

I would remove this function entirely, as well as removing `ariaExpanded` and `caretHtml`, and have `showFeedback` instead.